packages/components/src/internal/url/AppURLResolver.ts

```diff
@@ -306,3 +307,57 @@ export class SamplesResolver implements AppRouteResolver {
         }
     }
 }
+
+/**
+ * Resolves experiment runs to workflow jobs if appropriate
+ * /rd/run/14/... -> /workflow/14/...
+ * If this doesn't correspond to a job, the link won't resolve.
+ *
+ * Ideally we would resolve to the original URL if it's not a job, but since that's a link out to LKS
+ * it's not current supported by AppRouteResolvers.  Alternatively, and perhaps more ideally, we'd resolve
+ * to the lineage page for a sample, but the URL here doesn't have any info about the related entity.
+ */
+export class ExperimentRunResolver implements AppRouteResolver {
+
+    jobs: Set<number> // set of rowIds that are jobs
+
+    static createURL(rowId: string | number): AppURL {
+        return AppURL.create('rd','run', rowId);
+    }
+
+    constructor(jobs?: Set<number>) {
+        this.jobs = jobs !== undefined ? jobs : new Set();
+    }
+
+    fetch(parts: any[]): Promise<AppURL | boolean> {
```

Comment:
A few things to consider with this fetch:

1. I would recommend parsing the expected rowId so we can ensure it is a number (and then do the `isNan` check).
2. Looks like the `this.jobs` cache is not getting updated upon async resolution.
3. I recommend using an `async` method here to get rid of some of the boilerplate.

Here is this feedback rolled up (untested):

```ts
async fetch(parts: any[]): Promise<AppURL | boolean> {
    const rowIdIndex = 2;
    const rowId = parseInt(parts[rowIdIndex], 10);

    if (isNaN(rowId)) {
        // skip it
        return true;
    }
    if (this.jobs.has(rowId)) {
        // resolve it
        return AppURL.create('workflow', rowId);
    }

    try {
        const result = await selectRows({
            schemaName: SAMPLE_MANAGEMENT.JOBS.schemaName,
            queryName: SAMPLE_MANAGEMENT.JOBS.queryName,
            filterArray: [Filter.create('RowId', rowId)],
            columns: 'RowId',
        });

        if (Object.keys(result.models[result.key]).length) {
            // cache it
            this.jobs.add(rowId);

            // resolve it
            return AppURL.create('workflow', rowId);
        }

        // skip it
    } catch (e) {
        // skip it
    }

    // skip it
    return true;
}
```

Reply:
Thanks.  So efficient to create a cache and then not use it :-). I think this does have to return a Promise, though, so the bare `return X` statements don't work unless I'm missing something.

Reply:
The `async` portion of the definition means it will be returning a `Promise` of whatever type you're returning.